### ISSUE TITLE
False positive: no-ex-assign

### ISSUE THREAD

**Issue by RomkeVdMeulen** [buzinas/tslint-eslint-rules#54]
```
error (no-ex-assign) bootstrap.ts[34, 22]: invalid typeof comparison value
```

The line in question:

```
export function bootstrapComponents(): void {
	"use strict";

	bootstrap((aurelia: Aurelia) => { // this is line 34
		aurelia.use
			.defaultBindingLanguage()
			.defaultResources()
			.developmentLogging();

		aurelia.start().then(()  => aurelia.enhance({}, document.body));
	});
}
```

For one, the rule shouldn't apply here. Also, the error message doesn't match the rule description.

**Comment by buzinas** [comment]
Hey there, thanks for reporting it!

The string was incorrect, you're right, and there was already an issue reporting that. I've just fixed it.

Although, I couldn't simulate your error, in fact I've created a test putting your entire code there (https://github.com/buzinas/tslint-eslint-rules/commit/9da7ba8a45d172c5f5c04733da210a2b0a59d272) and it passes with no false positives. Can you please investigate further?

**Comment by RomkeVdMeulen** [comment]
It's really strange, but I'm still getting the same error after updating to the latest version of the rule. Maybe some kind of parse error? Here's the complete source:

```
[gulp-tslint] error (no-ex-assign) bootstrap.ts[23, 22]: do not assign to the exception parameter.
```

```
import {Aurelia} from "aurelia-framework";
import {bootstrap} from "aurelia-bootstrapper";
import {TemplateRegistryEntry, Loader} from "aurelia-loader";
import {TextTemplateLoader} from "aurelia-loader-default";

declare var frontendConfig: { translate: string, debug: boolean };

(<any>TextTemplateLoader).prototype.standardLoadTemplate = TextTemplateLoader.prototype.loadTemplate;
TextTemplateLoader.prototype.loadTemplate = function(loader: Loader, entry: TemplateRegistryEntry): Promise<any> {
	entry.address += "?translate=" + frontendConfig.language;
	return (<any>this).standardLoadTemplate(loader, entry);
};

export function bootstrapComponents(): void {
	"use strict";

	let paths: string[] = [];
	for (let i = 0; i < arguments.length; i++) {
		let component: string = arguments[i];
		paths.push("frontend/components/" + component + "/" + component);
	}

	bootstrap((aurelia: Aurelia) => { // line 23
		aurelia.use
			.defaultBindingLanguage()
			.defaultResources()
			.developmentLogging()
			.globalResources(paths);

		aurelia.start().then(()  => aurelia.enhance({}, document.body));
	});
}
```

**Comment by buzinas** [comment]
I could simulate the error now. I'll fix it, thanks!

**Comment by buzinas** [comment]
@nfriend

**Comment by Hufschmidt** [comment]
The following code also throws no-ex-assign on line 3 col 2

    import {Component} from '@angular/core';
    
    @Component({
    
    })
    export class App {
    }
    
with tslint 3.11 and tslint-eslint-rules 1.3.0

**Comment by Hufschmidt** [comment]
See PR, tested with all linked examples. Still triggering for actuall exception assignments, eg.

    import {Component} from '@angular/core';

    @Component({
    })
    export class MyApp {
      public hi() {
        try {
          console.log('Hello World');
        }
        catch (e) {
          // Hi
          let z = 'xxxx';
          e = 2;
          let x = e;
          z.toString();
          z = 'gfdg';
          console.log('Done!', x, z);
          console.log(e);
        }
      }
    }

Throws: src/app.ts[18, 11]: do not assign to the exception parameter.

**Comment by buzinas** [comment]
@Hufschmidt Unfortunately, the linting is not passing for the submitted PR.

**Comment by buzinas** [comment]
@zelein

**Comment by zelein** [comment]
Oh, thanks @buzinas I didn't realize this was here. Thanks for closing my issue.

**Comment by buzinas** [comment]
@Hufschmidt if you don't mind, I'll copy the ideas provided from your PR and submit one myself, so we can merge this.

**Comment by Hufschmidt** [comment]
Sure, go ahead. :)

**Comment by buzinas** [comment]
@Hufschmidt ty, just did it, the fix was released in v1.6.0
